cmd/snap/cmd_login.go

```diff
@@ -99,15 +100,26 @@ func requestLogin(email string) error {
 		return err
 	}
 
-	return requestLoginWith2faRetry(email, string(password))
+	// strings.TrimSpace needed because we get \r from the pty in the tests
+	return requestLoginWith2faRetry(email, strings.TrimSpace(string(password)))
 }
 
 func (x *cmdLogin) Execute(args []string) error {
 	if len(args) > 0 {
 		return ErrExtraArgs
 	}
 
-	err := requestLogin(x.Positional.Email)
+	email := x.Positional.Email
+	if email == "" {
+		fmt.Fprint(Stdout, "Please input your email address: ")
```

Comment:
i18n.G?
Also, given the other prompts are just `<thing you should enter>`, maybe this should be just `"email address"`?